synapseclient/core/utils.py

```diff
@@ -501,20 +551,41 @@ def from_unix_epoch_time(ms) -> datetime.datetime:
     return from_unix_epoch_time_secs(ms / 1000.0)
 
 
-def datetime_to_iso(dt, sep="T"):
-    # Round microseconds to milliseconds (as expected by older clients)
-    # and add back the "Z" at the end.
-    # see: http://stackoverflow.com/questions/30266188/how-to-convert-date-string-to-iso8601-standard
+def datetime_to_iso(
```

Comment:
Ah interesting, so there are places where we require the milliseconds, but just not for manifests?

Reply:
I'll double check on this logic. I know In the UI milliseconds are not enter-able. I'll check behavior through the `syn.set_annotations` functionality and align this logic to work the same way.

Reply:
ok so milliseconds are allowed. I updated the logic in https://github.com/Sage-Bionetworks/synapsePythonClient/pull/1030/commits/b73fd33bc4477996187f7fce209cf7dc1f5a8121 - This allows the date annotation to have milliseconds and when I am writing the manifest file only write the milliseconds if it's not 0.